discojs/src/privacy.ts

```diff
@@ -1,46 +1,39 @@
-import * as tf from '@tensorflow/tfjs'
+import * as tf from "@tensorflow/tfjs";
 
-import type { Task, WeightsContainer } from './index.js'
+import type { WeightsContainer } from "./index.js";
 
-/**
- * Add task-parametrized Gaussian noise to and clip the weights update between the previous and current rounds.
- * The previous round's weights are the last weights pulled from server/peers.
- * The current round's weights are obtained after a single round of training, from the previous round's weights.
- * @param updatedWeights weights from the current round
- * @param staleWeights weights from the previous round
- * @param task the task
- * @returns the noised weights for the current round
- */
-export function addDifferentialPrivacy (updatedWeights: WeightsContainer, staleWeights: WeightsContainer, task: Task): WeightsContainer {
-  const noiseScale = task.trainingInformation?.noiseScale
-  const clippingRadius = task.trainingInformation?.clippingRadius
+async function frobeniusNorm(weights: WeightsContainer): Promise<number> {
+  const squared = await weights
+    .map((w) => w.square().sum())
+    .reduce((a, b) => a.add(b))
+    .data();
+  if (squared.length !== 1) throw new Error("unexcepted weights shape");
 
-  const weightsDiff = updatedWeights.sub(staleWeights)
-  let newWeightsDiff: WeightsContainer
+  return Math.sqrt(squared[0]);
+}
 
-  if (clippingRadius !== undefined) {
-    // Frobenius norm
-    const norm = weightsDiff.frobeniusNorm()
+/** Scramble weights */
+export function addNoise(
+  scale: number,
+  weights: WeightsContainer,
+): WeightsContainer {
+  const deviation = Math.pow(scale, 2);
```

Comment:
naming: `variance` instead of std dev as it's squared?

Reply:
ho right, that's clearer. changed the "scale" argument to "deviation" too